core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala

```diff
@@ -410,47 +410,54 @@ sealed abstract class BitcoinTxBuilder extends TxBuilder {
-      case Nil => Success(currentLockTime)
-      case spendingInfo +: newRemaining =>
-        spendingInfo match {
-          case lockTime: LockTimeSpendingInfo =>
-            lockTime.scriptPubKey match {
-              case _: CSVScriptPubKey => loop(newRemaining, currentLockTime)
-              case cltv: CLTVScriptPubKey =>
-                val lockTime =
-                  if (cltv.locktime.toLong > UInt32.max.toLong || cltv.locktime.toLong < 0) {
-                    TxBuilderError.IncompatibleLockTimes
-                  } else Success(UInt32(cltv.locktime.toLong))
-                val result = lockTime.flatMap { l: UInt32 =>
-                  if (currentLockTime < l) {
-                    val lockTimeThreshold = tc.locktimeThreshold
-                    if (currentLockTime < lockTimeThreshold && l >= lockTimeThreshold) {
-                      //means that we spend two different locktime types, one of the outputs spends a
-                      //OP_CLTV script by block height, the other spends one by time stamp
+        currentLockTimeOpt: Option[UInt32]): Try[UInt32] =
+      remaining match {
+        case Nil =>
+          Success(currentLockTimeOpt.getOrElse(TransactionConstants.lockTime))
+        case spendingInfo +: newRemaining =>
+          spendingInfo match {
+            case lockTime: LockTimeSpendingInfo =>
+              lockTime.scriptPubKey match {
+                case _: CSVScriptPubKey =>
+                  loop(newRemaining, currentLockTimeOpt)
+                case cltv: CLTVScriptPubKey =>
+                  val lockTime =
+                    if (cltv.locktime.toLong > UInt32.max.toLong || cltv.locktime.toLong < 0) {
                       TxBuilderError.IncompatibleLockTimes
-                    } else Success(l)
-                  } else Success(currentLockTime)
-                }
-                result
-            }
-          case p2sh: P2SHSpendingInfo =>
-            loop(p2sh.nestedSpendingInfo +: newRemaining, currentLockTime)
-          case p2wsh: P2WSHV0SpendingInfo =>
-            loop(p2wsh.nestedSpendingInfo +: newRemaining, currentLockTime)
-          case conditional: ConditionalSpendingInfo =>
-            loop(conditional.nestedSpendingInfo +: newRemaining,
-                 currentLockTime)
-          case _: P2WPKHV0SpendingInfo |
-              _: UnassignedSegwitNativeUTXOSpendingInfo | _: P2PKSpendingInfo |
-              _: P2PKHSpendingInfo | _: MultiSignatureSpendingInfo |
-              _: EmptySpendingInfo =>
-            // none of these scripts affect the locktime of a tx
-            loop(newRemaining, currentLockTime)
-        }
-    }
+                    } else Success(UInt32(cltv.locktime.toLong))
+                  val result = lockTime.flatMap { l: UInt32 =>
+                    currentLockTimeOpt match {
+                      case Some(currentLockTime) =>
+                        if (currentLockTime < l) {
```

Comment:
Doesn't this assume that `l` is a block time and not a block height, if so wouldn't this break when a block height is given for `l` and `currentLockTime` is a block time